lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -876,6 +884,12 @@ def _start_opensearch(self, event: _StartOpenSearch) -> None:  # noqa: C901
             )
             self._post_start_init(event)
         except (OpenSearchHttpError, OpenSearchStartTimeoutError, OpenSearchNotFullyReadyError):
+            if "data" not in deployment_desc.config.roles:
+                self.status.set(BlockedStatus(PClusterNoDataNode))
+                self.node_lock.release()
+            if self.opensearch_peer_cm.is_provider():
```

Comment:
I believe this should be `is_provider(typ="main")`

Reply:
done.